src/openapi/streaming/connection/transport/websocket-transport.ts

```diff
@@ -377,6 +390,53 @@ class WebsocketTransport implements StreamingTransportInterface {
         }
     }
 
+    private startInactivityFinder = () => {
+        if (this.inactivityFinderRunning) {
+            log.warn(
+                LOG_AREA,
+                'Starting inactivityFinder when already started',
+            );
+        }
+        this.inactivityFinderRunning = true;
+        this.inactivityFinderNextUpdateTimeoutId = window.setTimeout(
+            this.onInactivityFinderUpdate,
+            MAX_INACTIVITY_WAIT_TIME,
+        );
+    };
+
+    private stopInactivityFinder = () => {
+        if (this.inactivityFinderNextUpdateTimeoutId) {
+            clearTimeout(this.inactivityFinderNextUpdateTimeoutId);
+            this.inactivityFinderNextUpdateTimeoutId = null;
+        }
+        this.inactivityFinderRunning = false;
+    };
+
+    private onInactivityFinderUpdate = () => {
+        if (!this.inactivityFinderRunning) {
+            return;
+        }
+
+        const now = Date.now();
+        if (
+            !this.isReconnectPending &&
+            this.lastMessageTime &&
+            now - this.lastMessageTime >= MAX_INACTIVITY_WAIT_TIME
+        ) {
+            // no message on tranport since MAX_INACTIVITY_WAIT_TIME duration so reconnect
+            log.info(
+                LOG_AREA,
+                'Inactivity finder reconnecting since no message recieved since 3 seconds',
+            );
+            this.reconnect();
+        }
+
+        this.inactivityFinderNextUpdateTimeoutId = window.setTimeout(
+            this.onInactivityFinderUpdate,
+            MAX_INACTIVITY_WAIT_TIME,
+        );

```

Comment:
One thought here,  seems like the `lastMessageTime` is not reset anywhere, let's say that we detected inactivity so we reconnect and schedule next check after 3s. I'm wondering if it is possible to jump to this method again after reconnection without `lastMessageTime` being updated which would mean we reconnect immediately again.

Reply:
`lastMessageTime` only updates when we get snapshot, deltas or heartbeat message.
and yes if we still not received any message during waiting for 3 seconds it should reconnect after that and its the expected behavior

Reply:
yeah, but in case the lastMessageTime is not updated by any snapshot/heartbeat etc. after reconnection, we will immediately reconnect since we will have the old lastMessageTime - ofc only if such a scenario is even possible.